history.js

```diff
@@ -240,7 +240,7 @@
          */
         "state": {
             get: function() {
-                return stateStorage[windowLocation.href] || null;
+                return typeof stateStorage[windowLocation.href] === 'object' ? JSON.parse(JSON.stringify(stateStorage[windowLocation.href])) : (stateStorage[windowLocation.href] || null);
```

Comment:
Oh I see you integrated JSON object to IE7- version. Than OK - I'll make some changes there and reformat it to clearer look